### ISSUE TITLE
How to custom header during handshake?

### ISSUE THREAD

**Issue by roaraurora** [stomp-js/ng2-stompjs#83]
Hi, I am looking for a way to add custom header for **websocket handshake**,  not normal websocket frame, so is there any way to accomplish it? very thanks.

**Comment by kum-deepak** [comment]
I guess you meant the HTTP request. Unfortunately the WebSocket standards do not provide a way to add headers there.

**Comment by kum-deepak** [comment]
Are you trying to use it with Spring based back-end?

**Comment by roaraurora** [comment]
Yes,I am using it with spring framwork.And if there is nothing I can do within transportion layer,I guess it time to close this issue @kum-deepak

**Comment by kum-deepak** [comment]
I do not program in Java/Spring. Based on issues reported, I believe that handling Authentication in Spring Application relies on cookies/headers - even when using WebSockets (plain or with STOMP). Which might work with a regular web site. However, when using with no-cookie environment like Angular, it becomes challenging.

**Comment by roaraurora** [comment]
Currently I am using JWT as authorization, and I realized that the only way to use it with websocket is to intercet and handle first connect command frame send from client. Thanks for your opinion and helpful thoughts, I decide to compromise with this solution metioned above.

**Comment by kum-deepak** [comment]
This question has been asked multiple times. Will it be possible for you to share a minimal working Java application that demonstrates the authentication using JWT?

Many thanks in advance!
